### PR TITLE
Added support for pruning of torch.nn.functional.pad operation

### PR DESCRIPTION
**Pull request by nikita-savelyevv**
### Changes

- Made pad operation accept pruned input

### Reason for changes

Several models were affected

### Related tickets

86890

### Tests

Added a test model containing various padding operations
image

### CODE REVIEW

File: nncf/common/graph/layer_attributes.py [hunk 1]
```diff
@@ -227,30 +228,24 @@ def get_target_dim_for_compression(self) -> int:
         return 0
 
 
+@dataclass
```
Comment (nikita-savelyevv) [inline nncf/common/graph/layer_attributes.py:231]: Decided to wrap with `dataclass` for convenience

File: nncf/torch/dynamic_graph/layer_attributes_handlers.py [hunk 2]
```diff
@@ -44,12 +44,14 @@
...
 OP_NAMES_REQUIRING_MODULE_ATTRS = [
     v.op_func_name for v in NNCF_MODULES_DICT] + list(PTGroupNormMetatype.get_all_aliases())
-OP_NAMES_REQUIRING_ATTRS_FROM_ARGS_KWARGS = list(
-    PTTransposeMetatype.get_all_aliases() + PTGatherMetatype.get_all_aliases()
-)
+
 TRANSPOSE_OP_NAMES = ['transpose', 'transpose_']
 PERMUTE_OP_NAMES = ['permute']
 GETITEM_OP_NAMES = ['__getitem__']
+GETITEM_PAD_NAMES = PTPadMetatype.get_all_aliases()
```
Comment (nikita-savelyevv) [inline nncf/torch/dynamic_graph/layer_attributes_handlers.py:51]: Decided to change the definition of this list, because later only the function names from `TRANSPOSE_OP_NAMES`, `PERMUTE_OP_NAMES`, `GETITEM_OP_NAMES` will pass the check inside `get_layer_attributes_from_module` and all function aliases from corresponding metatypes are not needed.

File: nncf/torch/graph/operator_metatypes.py [hunk 3]
```diff
@@ -931,9 +931,6 @@ def get_operator_metatypes() -> List[Type[OperatorMetatype]]:
     return list(PT_OPERATOR_METATYPES.registry_dict.values())
 
 
-OP_NAMES_REQUIRING_ATTRIBUTES_FROM_ARGS_KWARGS = [PTTransposeMetatype.get_all_aliases()]
```
Comment (nikita-savelyevv) [inline nncf/torch/graph/operator_metatypes.py:934]: Not referenced anywhere, removed

**Comment by daniil-lyakhov** [comment]
Thank you for the contribution! Do you have a plan to support pad operation in TF as well? If not - please make a ticket for that

**Comment by nikita-savelyevv** [comment]
> Thank you for the contribution! Do you have a plan to support pad operation in TF as well? If not - please make a ticket for that

Done: 106749

**Review by ljaljushkin — APPROVED** [review]

**Review by daniil-lyakhov — APPROVED** [review]
Good job 👍

**Comment by nikita-savelyevv** [comment]
Build 292 fails for some models, as does the develop build 291. Since there are no regressions compared to the develop, I suggest it is safe to merge this PR.
